pallets/collator-selection/src/lib.rs

```diff
@@ -499,29 +512,96 @@ pub mod pallet {
 			collators.extend(candidates.into_iter().collect::<Vec<_>>());
 			collators
 		}
-		/// Kicks out and candidates that did not produce a block in the kick threshold.
+
+		/// Removes collators with unsatisfactory performance
+		/// Returns the removed AccountIds
 		pub fn kick_stale_candidates(
 			candidates: Vec<CandidateInfo<T::AccountId, BalanceOf<T>>>,
 		) -> Vec<T::AccountId> {
-			let now = frame_system::Pallet::<T>::block_number();
-			let kick_threshold = T::KickThreshold::get();
-			candidates
+			// 0. Storage reads and precondition checks
+			if candidates.is_empty() {
+				return Vec::new(); // No candidates means we're running invulnerables only
+			}
+			let percentile_for_kick = T::PerformancePercentileToConsiderForKick::get();
+			if percentile_for_kick == 0 {
+				return Vec::new(); // Selecting 0-th percentile disables kicking. Upper bound check in fn build()
```

Comment:
How about `return None`, and the return can be defined as `Option<Vec<T::AccountId>>`?

Reply:
good point, thanks!
I didnt think of changing the signature of kick_stale_candidates

Reply:
added in [341b9e9](https://github.com/Manta-Network/Manta/pull/358/commits/341b9e95bce0a18f9c7afeed791013379281a2a1)